### PR TITLE
[SYNPY-1417] Updates for the annotation model

### PR DESCRIPTION
**Pull request by BryanFauble**
**Problem:**

1. When using the annotations field you had to first set it to a dict to use it since the default was None.
2. When using the annotations field it was not intuitive to set it to an empty dict to delete all annotations on an entity.

**Solution:**

1. Updating the default for the annotation field to be an empty dict.
2. Updating the logic around the annotation field such that if it is Empty or None it will be treated as a delete. It still means though that to delete annotations for an entity that is already stored in Synapse you'll need to do a "destructive" store (Meaning you `.get()` it then `.store()` it.

**Testing:**

1. Verified with new and existing integration tests

### CODE REVIEW

File: synapseclient/models/services/storable_entity_components.py [hunk 1]
```diff
@@ -186,7 +186,7 @@ async def _store_activity_and_annotations(
     )
     if (
         hasattr(root_resource, "annotations")
-        and root_resource.annotations is not None
+        and (root_resource.annotations or last_persistent_instance)
```
[the remark this answers is not in the document]
Comment (BryanFauble) [reply synapseclient/models/services/storable_entity_components.py:189]: To clear up a few items:

1. This script is only called from items in the new models directory: https://github.com/Sage-Bionetworks/synapsePythonClient/blob/develop/synapseclient/models/project.py - This does not include items returned by `syn.get`. Those are still returning the "old" Entities

> `hasattr(root_resource, "annotations")`
I left this in as a guard statement in several places in this file for the different attached items to an Entity. I have plans to revamp the saving of these resources so they are all handled in a single HTTP call in `/bundle`.

> For the if statement, I understand that we would like to check the existence (non-empty) of root_resource.annotations and the existence of last_persistent_instance, but the latter has been checked in the third statement i.e last_persistent_instance is None ). Can you double check this? Thanks

For the if statement I need to handle for a few things:

1. You have just created a Project and are assigning NO annotations -> Do not store annotations
2. You have just created a Project and are assigning annotations -> Store the annotations
3. You already created a Project, but are updating some other field -> Do not store annotations as they did not change
4. You already created a Project, but are updating the annotations -> Store the update
5. You already created a Project, but are deleting all annotations -> Store the update

[the remark this answers is not in the document]
Comment (danlu1) [reply synapseclient/models/services/storable_entity_components.py:189]: Thanks for clarifying. You have just created a Project and are assigning NO annotations -> Do not store annotations
You have just created a Project and are assigning annotations -> Store the annotations
You already created a Project, but are updating some other field -> Do not store annotations as they did not change
You already created a Project, but are updating the annotations -> Store the update
You already created a Project, but are deleting all annotations -> Store the update

[the remark this answers is not in the document]
Comment (danlu1) [reply synapseclient/models/services/storable_entity_components.py:189]: Would root_resource.annotations return an empty dictionary if no annotations are assigned?

[the remark this answers is not in the document]
Comment (BryanFauble) [reply synapseclient/models/services/storable_entity_components.py:189]: The default value is going to be `{}`, but you could also assign the value to None and clear annotations out. like:
```
loaded_project.annotations = {}
loaded_project.annotations = None
```

[the remark this answers is not in the document]
Comment (danlu1) [reply synapseclient/models/services/storable_entity_components.py:189]: Gotcha. Thanks for clarifying. It looks good and I'm going to approve it.

**Review by danlu1 — APPROVED** [review]

**Pull request merged**